File: InMotionGIT.Common/Helpers/MultiCompany.vb
Namespace Helpers

    Public Class MultiCompany

#Region "Field"

        Public Property Identification As Integer
        Public Property Name As String

#End Region

#Region "Method"

        Public Shared Function GetUserInfo(ByVal nCompanyId As Short) As Object
            Dim clsConfig As VisualTimeConfig
            Dim companyName As String = ""
            Dim companyUser As String = ""
            Dim companyPassword As String = ""
            Dim arrResult(2) As Object

            arrResult(0) = String.Empty
            arrResult(1) = String.Empty
            arrResult(2) = String.Empty
            clsConfig = New VisualTimeConfig
            If clsConfig.GetCompanySettings(nCompanyId, companyName, companyUser, companyPassword) Then
                arrResult(0) = companyName
                arrResult(1) = companyUser
                arrResult(2) = companyPassword
            End If

            GetUserInfo = arrResult
        End Function

        Public Shared Function IsMultiCompany() As Boolean
            Dim sMultiCom As String = String.Empty
            With New Values
                sMultiCom = .insGetSetting("MultiCompany", "No", "Database")
            End With
            Return (String.Compare(sMultiCom, "Yes", True) = 0)
        End Function

        Public Shared Function MultiCompanyList() As DataTable
            Dim clsConfig As VisualTimeConfig
            Dim companyName As String = String.Empty
            Dim intIndex As Integer

            ' Definicion del DataTable
            Dim List As New DataTable("List")
            Dim id As New DataColumn("id")
            Dim name As New DataColumn("name")

            ' Creacion del DataTable
            List.Columns.Add(id)
            List.Columns.Add(name)

            clsConfig = New VisualTimeConfig

            For intIndex = 1 To 20
                If clsConfig.GetCompanySettings(intIndex, companyName, "", "") Then
                    List.Rows.Add(intIndex.ToString, companyName)
                End If
            Next intIndex

            Return List
        End Function

#End Region

    End Class

End Namespace

File: Common/InMotionGIT.Common/Helpers/MultiCompany.cs
using System.Data;

namespace InMotionGIT.Common.Helpers
{

    public class MultiCompany
    {

        #region Field

        public int Identification { get; set; }
        public string Name { get; set; }

        #endregion

        #region Method

        public static object GetUserInfo(short nCompanyId)
        {
            object GetUserInfoRet = default;
            VisualTimeConfig clsConfig;
            string companyName = "";
            string companyUser = "";
            string companyPassword = "";
            var arrResult = new object[3];

            arrResult[0] = string.Empty;
            arrResult[1] = string.Empty;
            arrResult[2] = string.Empty;
            clsConfig = new VisualTimeConfig();
            if (clsConfig.GetCompanySettings(nCompanyId, ref companyName, ref companyUser, ref companyPassword))
            {
                arrResult[0] = companyName;
                arrResult[1] = companyUser;
                arrResult[2] = companyPassword;
            }

            GetUserInfoRet = arrResult;
            return GetUserInfoRet;
        }

        public static bool IsMultiCompany()
        {
            string sMultiCom = string.Empty;
            {
                var withBlock = new Values();
                sMultiCom = withBlock.insGetSetting("MultiCompany", "No", "Database");
            }
            return string.Compare(sMultiCom, "Yes", true) == 0;
        }

        public static DataTable MultiCompanyList()
        {
            VisualTimeConfig clsConfig;
            string companyName = string.Empty;
            int intIndex;

            // Definicion del DataTable
            var List = new DataTable("List");
            var id = new DataColumn("id");
            var name = new DataColumn("name");

            // Creacion del DataTable
            List.Columns.Add(id);
            List.Columns.Add(name);

            clsConfig = new VisualTimeConfig();

            for (intIndex = 1; intIndex <= 20; intIndex++)
            {
                string argcompanyUser = "";
                string argcompanyPassword = "";
                if (clsConfig.GetCompanySettings((short)intIndex, ref companyName, ref argcompanyUser, ref argcompanyPassword))
                {
                    List.Rows.Add(intIndex.ToString(), companyName);
                }
            }

            return List;
        }

        #endregion

    }

}